### DISCUSSION TITLE
Duda sobre tiempo de ejecución

### DISCUSSION THREAD

**Discussion by gabo1243** [IIC2133-PUC/2024-1#98] [Tarea 1] [answered]
Hola, estaba corriendo los tests de la parte 4 (los nuevos) y me pasó que el test 4-x_range-wikichileans solo corría exitosamente si le ponía valgrind antes del comando, sino parece que se demora mucho o simplemente no termina. Es por eso que pregunto si valgrind hace que se corra más rápido el código o no es así. (sin valgrind me llega a escribir hasta la linea 577 del .txt y no más, pero con valgrind me escribe las 61105 en uno o dos seg).

Gracias de antemano

**Comment by sebacarrascop** [comment, marked as answer]
No!, hay algo extraño, en general valgrind hace más lenta la ejecución ya que realiza una simulación del procesador y ejecuta el programa sobre un modelo virtual de la maquina.
Prueba haciendo un make clean && make, si sigue el problema debe ser algo del código, saludos!

**Comment by gabo1243** [comment]
Hola, graciass, no sé si era eso, pero se solucionó al cambiar tambien unos errores de valores no inicializados (Conditional jump or move depends on uninitialised value(s)).
